node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1232,7 +1265,14 @@ async fn fetch_blocks_and_state_since_genesis(ctx: &ChainSyncContext<'_>) -> Res
     let latest_height_requested: Arc<AtomicU64> = Arc::new(AtomicU64::new(0));
 
     let mut workers: FuturesUnordered<_> = (0..ctx.config.max_parallel_block_fetches())
-        .map(|worker_id| fetch_block_worker(worker_id, latest_height_requested.clone(), ctx))
+        .map(|worker_id| {
+            fetch_block_worker(
+                worker_id,
+                latest_height_requested.clone(),
+                validator_weights.clone(),
```

Comment:
This won't be correct either - validators change and here we would use the same validator set throughout the whole process.